lib/Sema/TypeCheckDecl.cpp

```diff
@@ -482,6 +482,47 @@ static void checkInheritanceClause(
   }
 }
 
+// Check for static properties that produce empty option sets
+static void checkForEmptyOptionSet(const VarDecl *VD, TypeChecker &tc) {
+  if (!VD->isStatic() || !VD->isLet())
+    return;
+  
+  auto DC = VD->getDeclContext();
+  
+  auto *optionSetProto = tc.Context.getProtocol(KnownProtocolKind::OptionSet);
+  auto protocolConformance = tc.containsProtocol(DC->getSelfTypeInContext(), optionSetProto, DC, /*Flags*/None);
```

Comment:
Nitpick: We have an 80-column limit; please wrap and align to the first open paren.

Also nitpick: I'd probably just use `bool` here, but if you want to use `auto` please pick a name that sounds like a true/false value, like "conformsToOptionSet".

Reply:
Will do. It is actually returning an Optional<swift::ProtocolConformanceRef>. Should I check if the optional is empty to produce a bool instead?

Reply:
Would it be ok to use a double negation to convert it into a bool?

Reply:
Ah, right. I don't think we use the double negation thing too much in this codebase; an explicit call to Optional::hasValue would be most clear, but also longer.

Now that I know what the type is, leaving it the way it is is also okay. It does, in fact, represent a possibly-absent conformance.